tests/xrt/perf_IOPS/xcl_api_iops.cpp

```diff
@@ -177,79 +189,76 @@ int testSingleThread(xclDeviceHandle handle, xuid_t uuid, int bank)
     return 0;
 }
 
-void *runTestThread(void *data)
+void *runTestThread(arg_t &arg)
 {
-    arg_t *arg = (arg_t *)data;
     std::vector<std::shared_ptr<task_info>> cmds;
     /* The command would incease */
 
-    fillCmdVector(arg->handle, cmds, arg->bank, arg->queueLength);
+    fillCmdVector(arg.handle, cmds, arg.bank, arg.queueLength);
 
-    pthread_barrier_wait(&barrier);
+    barrier.wait();
 
-    //wait start from main thread
+    double duration = runTest(arg.handle, cmds, arg.total, arg);
 
-    double duration = runTest(arg->handle, cmds, arg->total, arg);
-
-    pthread_barrier_wait(&barrier);
+    barrier.wait();
 
     for (auto& cmd : cmds) {
-        xclFreeBO(arg->handle, cmd->boh);
+        xclFreeBO(arg.handle, cmd->boh);
         munmap(cmd->ecmd, 4096);
-        xclFreeBO(arg->handle, cmd->exec_bo);
+        xclFreeBO(arg.handle, cmd->exec_bo);
     }
 }
 
-/* let's start from test two threads */
 int testMultiThreads(xclDeviceHandle handle, xuid_t uuid, int bank,
         int threadNumber, int queueLength, unsigned int total)
 {
-    pthread_t *tids = (pthread_t *)malloc(threadNumber * sizeof(pthread_t));
-    arg_t *arg[threadNumber];
+    std::thread threads[threadNumber];
+    std::vector<arg_t> arg(threadNumber);
 
     if (xclOpenContext(handle, uuid, 0, true))
         throw std::runtime_error("Cound not open context");
 
-    pthread_barrier_init(&barrier, NULL, threadNumber+1);
+    barrier.init(threadNumber + 1);
 
     for (int i = 0; i < threadNumber; i++) {
-        arg[i] = (arg_t *)malloc(sizeof(arg_t));
-        arg[i]->thread_id = i;
-        arg[i]->bank = bank;
-        arg[i]->handle = handle;
-        arg[i]->queueLength = queueLength;
-        arg[i]->total = total;
-        pthread_create(&tids[i], NULL, runTestThread, arg[i]);
+        arg[i].thread_id = i;
+        arg[i].bank = bank;
+        arg[i].handle = handle;
+        arg[i].queueLength = queueLength;
+        arg[i].total = total;
+        threads[i] = std::move(std::thread(runTestThread, std::ref(arg[i])));
```

Comment:
I guess the `std::move()` is redundant.
Also, in modern C++ you do not need these `std::ref` and old API.
You can just use a lambda and a normal API.
```c++
threads[i] = std::move { [&] { runTestThread(i, bank, handle, queueLength, total); } };
```
Modern C++ is more like Python than C and `void**`. :-)

Reply:
Good suggestion. :)